net/tlsdial/tlsdial.go

```diff
@@ -52,6 +63,9 @@ func Config(host string, base *tls.Config) *tls.Config {
 		conf = base.Clone()
 	}
 	conf.ServerName = host
+	if len(serverHostOverride()) != 0 {
+		conf.ServerName = serverHostOverride()
+	}
```

Comment:
But this tlsdial.Config func is used for more than just dialing the control plane. There are ~4 different types of services it's used for, so 1 env var can't work for all of them.